src/FSharpPlus/Extensions/ResizeArray.fs

```diff
@@ -86,3 +86,15 @@ module ResizeArray =
         let (x, y) = ResizeArray (), ResizeArray ()
         Array.iter (mapper >> function Choice1Of2 e -> x.Add e | Choice2Of2 e -> y.Add e) source
         x.ToArray (), y.ToArray ()
+    
+    
+    /// <summary>
+    // Zip safely two ResizeArrays. If one ResizeArray is shorter, excess elements are discarded from the right end of the longer ResizeArray. 
+    /// </summary>
+    /// <param name="a1">First input ResizeArray.</param>
+    /// <param name="a2">Second input ResizeArray.</param>
+    /// <returns>ResizeArray with corresponding pairs of input ResizeArrays.</returns>
+    let zipShortest (a1: ResizeArray<'T1>) (a2: ResizeArray<'T2>) =
```

Comment:
@gusty , I wonder if the naming is right. FSharp.Core doesn't have a `zip` function for ResizeArray, so we could use just `zip` here. However, `FSharpx.Collections` have `ResizeArray.zip` that fails when arrays are of different length.

Reply:
You are right, let's stay on more time on the safe side: leave as it is now.